test/corpus/listitem.txt

```diff
@@ -0,0 +1,447 @@
+        (word))
+      (taglink
+        (word))
+      (word))
+    (line
+      (argument
+        (word))
+      (word))
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (taglink
+          (word)))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (taglink
+          (word)))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (taglink
+          (word)))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word))
+      (line)
+      (line
+        (argument
+          (word))
+        (word)))))
+
+================================================================================
+listitem tricky
+================================================================================
+
+- x - x
+
+-x -x
+
+- - x -x
+- -x - x
+- -
+
+
+--------------------------------------------------------------------------------
+
+(help_file
+  (block
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word)
+        (word))
+      (line)))
+  (block
+    (line
+      (word)
+      (word)))
+  (block
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word)
+        (word))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word)
+        (word))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word))
+      (line))))
+
+
+================================================================================
+numbered listitem
+================================================================================
+1. item 1 text
+   continued line
+2. 2 item text
+3. item
+  - nested
+    continued line
+    1. nested
+       continued line
+ 
+999. new block
+
+9999. number too big
+ 
+- new block
+
+
+--------------------------------------------------------------------------------
+
+(help_file
+  (block
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word)
+        (word))
+      (line)
+      (line
+        (word)
+        (word)))
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word)
+        (word))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word))
+      (line))
+    (line_li
+      (prefix)
+      (line
+        (word))
+      (line)
+      (line
+        (word)
+        (word)))
+    (line_li
+      (prefix)
+      (line
+        (word))
+      (line)
+      (line
+        (word)
+        (word))))
+  (block
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word))
+      (line)))
+  (block
+    (line
+      (word)
+      (word)
+      (word)
+      (word)))
+  (block
+    (line_li
+      (prefix)
+      (line
+        (word)
+        (word))
+      (line))))
+
+================================================================================
+NOT listitem (FIXME)
+================================================================================
+block1
+- block1
+
+block1 line1
+1. block1 line2
```

Comment:
Initially thought that this isn't wanted, but it's more common that it *is* wanted. So will change this test to "expected" instead of "FIXME".

This matches the behavior of lists in markdown.